src/header/NamespaceDropdown/NamespaceDropdown.tsx

```diff
@@ -1,18 +1,46 @@
 import { Menu, Icon } from 'fundamental-react';
-import { Link } from 'react-router-dom';
+import { useRecoilValue } from 'recoil';
+import { Link, useMatch } from 'react-router-dom';
 import { useTranslation } from 'react-i18next';
 
+import { namespacesState } from 'state/namespacesAtom';
 import { useUrl } from 'hooks/useUrl';
-import { NamespacesState } from 'state/namespacesAtom';
 
 import './NamespaceDropdown.scss';
 
+const Namespaces = () => {
+  const namespaces = useRecoilValue(namespacesState);
+  const { clusterUrl, namespace, namespaceUrl } = useUrl();
+
+  const { resourceType = '' } =
+    useMatch({
+      path: '/cluster/:cluster/namespaces/:namespace/:resourceType',
+      end: false,
+    })?.params ?? {};
+
+  const switchNamespace = (namespaceName: string) => {
+    if (!namespace) return clusterUrl(`namespaces/${namespaceName}`);
+
+    return namespaceUrl(resourceType, {
+      namespace: namespaceName,
+    });
```

Comment:
do we really need that `if` here?
won't just
```suggestion
    return namespaceUrl(resourceType, {
      namespace: namespaceName,
    });
```
work?
unless that adds `undefined` to the url, but that should be handled in `useUrl`
(use `path ?? ''` instead of just `path` in the url string)